mat/cholesky.go

```diff
@@ -666,31 +663,31 @@ func (c *Cholesky) SymRankOne(orig *Cholesky, alpha float64, x Vector) (ok bool)
 	}
 	umat := c.chol.mat
 	stride := umat.Stride
+	workData := getFloats(umat.N*umat.N, false)
+	defer putFloats(workData)
+	copy(workData, umat.Data)
```

Comment:
This relies on `umat.Stride == umat.N`. I think that currently this is always true, I hope it will not change in the future because this assumption is hard to spot. An alternative would be to use `workData` to create a `TriDense` and then use its `Copy` here and .

It would be also good to say in the doc comment that the receiver won't be modified if the update fails.

Reply:
This reliance on stride was my concern.The `chol` matrix is never touched by others so I think it's probably safe, but it would be worth putting a comment on the field to this effect, if we want to do this. The `TriDense.Copy` approach is safe.

Reply:
fair enough, and reworked to be a smaller change as well